app/models/dor_object_workflow_status.rb

```diff
@@ -0,0 +1,18 @@
+class DorObjectWorkflowStatus
+  attr_reader :pid
+
+  ##
+  # @param [String] pid in format "druid:abc123def4567"
+  def initialize(pid)
+    @pid = pid
+  end
+
+  ##
+  # @return [Boolean]
+  def can_open_version?
+    return false unless Dor::WorkflowService.get_lifecycle('dor', pid, 'accessioned')
+    return false if Dor::WorkflowService.get_active_lifecycle('dor', pid, 'submitted')
+    return false if Dor::WorkflowService.get_active_lifecycle('dor', pid, 'opened')
+    true
+  end
```

Comment:
Maybe this could be simplified?  Like:
```ruby
Dor::WorkflowService.get_lifecycle('dor', pid, 'accessioned')  &&
!( Dor::WorkflowService.get_active_lifecycle('dor', pid, 'submitted') ||
   Dor::WorkflowService.get_active_lifecycle('dor', pid, 'opened'))
```